lighthouse-core/gather/computed/metrics/first-contentful-paint.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const MetricArtifact = require('./metric');
+const LHError = require('../../../lib/errors');
+
+class FirstContentfulPaint extends MetricArtifact {
+  get name() {
+    return 'FirstContentfulPaint';
+  }
+
+  /**
+   * @param {LH.Artifacts.MetricComputationData} data
+   * @return {Promise<LH.Artifacts.Metric>}
+   */
+  computeObservedMetric(data) {
+    const {traceOfTab} = data;
+    if (!traceOfTab) {
```

Comment:
hmm, this is kind of unfortunate to check here, but it looks like at least tsc alerts you based on `MetricComputationData`? Should we always require a trace for ComputedMetric? I assume you made `traceOfTab` optional for a reason, though

Reply:
yeah I split MetricComputationData into two so we don't have to do this